src/webview_provider/insightsPanel.ts

```diff
@@ -376,13 +392,153 @@ export class InsightsPanel extends AltimateWebviewProvider {
     });
   }
 
+  private async selectFiles(
+    syncRequestId: string | undefined,
+    { filters = {}, canSelectMany = false }: SelectFilesProps,
+  ) {
+    const openDialog = await window.showOpenDialog({
+      filters,
+      canSelectFolders: false,
+      openLabel: "Select",
+      canSelectFiles: true,
+      canSelectMany,
+    });
+    if (openDialog === undefined || openDialog.length === 0) {
+      this.dbtTerminal.debug("InsightsPanel", "opendialog cancelled");
+      this._panel!.webview.postMessage({
+        command: "response",
+        args: {
+          syncRequestId,
+          body: { error: "File not selected" },
+          status: false,
+        },
+      });
+      return;
+    }
+
+    this._panel!.webview.postMessage({
+      command: "response",
+      args: {
+        syncRequestId,
+        body: { path: openDialog.map((i) => i.fsPath) },
+        status: true,
+      },
+    });
+  }
+
+  private async altimateScan(
+    syncRequestId: string | undefined,
+    args: AltimateConfigProps,
+  ) {
+    let isInstalled = false;
```

Comment:
i can see 3 different operations happening in this method,
- checking datapilot installation
- installing datapilot
- running project healthcheck

Can you extract these 3 to separate functions so it can be reused if needed and better readability?

Reply:
its mostly like that. it just looks big due to error handling. All the business logic related to operations are happening inside `dbtProjectContainer.ts` and `dbtProject.ts`